### PR TITLE
Switch build backend to hatchling

### PR DESCRIPTION
**Pull request by hoechenberger**
### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

cc @ofek in case you're interested :)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -83,19 +84,21 @@ homepage = "https://mne.tools/mne-bids-pipeline"
...
+raw-options = { version_scheme = "release-branch-semver" }
 
-[tool.setuptools.package-data]
-"mne_bids_pipeline.steps.freesurfer.contrib" = ["version"]
+[tool.hatch.build]
+exclude = [
+    "/.*",
+    "/codecov.yml",
+    "**/tests",
+    "/docs",
```
Comment (hoechenberger) [inline pyproject.toml:96]: @ofek I'm explicitly excluding `/docs` here, but `docs/source/examples/gen_examples.py` is still being packaged when I build the sdist. Any idea why that's happening?

Comment (hoechenberger) [reply pyproject.toml:96]: Ah! There's an entry in `.gitignore`:

```
!docs/source/examples/gen_examples.py
```

Wonder why we have that one … but this explains it.

**Comment by hoechenberger** [comment]
@larsoner i think this is good to go

**Comment by larsoner** [comment]
Nice @hoechenberger !

**Pull request merged**